src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.ts

```diff
@@ -76,9 +87,49 @@ describe("Carousel", () => {
         });
 
         it("renders only one active carousel item", () => {
-            const activeItems = carouselWrapper.find(CarouselItem).filterWhere(c => c.prop("active"));
+            const activeItems = carouselWrapper.find(CarouselItem).filterWhere((c) => c.prop("active"));
 
             expect(activeItems.length).toBe(1);
         });
+
+        it("renders carousel controls", () => {
+            const carouselControls = carousel.find(CarouselControl);
+
+            expect(carouselControls.length).toBe(2);
+            expect(carouselControls.at(0).props().direction).toBe("left");
+            expect(carouselControls.at(1).props().direction).toBe("right");
+        });
+
+        it("switches to the next image when the right control is clicked", () => {
+            const carouselControls = carousel.find(CarouselControl);
+            const rightControl = carouselControls.at(1);
+
+            expect(carousel.state().activeIndex).toBe(0);
+
+            rightControl.simulate("click");
+
+            expect(carousel.state().activeIndex).toBe(1);
+
+            rightControl.simulate("click");
+
+            expect(carousel.state().activeIndex).toBe(0);
+            // Test incomplete: not checking if carousel items are updated. Failed to successfully test
```

Comment:
Fixed, The carouselItems are not a reference, so they should be found again after the test.